### PR TITLE
[NTOS:KE] KAFFINITY fixes

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

- Fix some wrong types, which should be KAFFINITY
- Fix wrong 64 bit affinity value

### CODE REVIEW

File: ntoskrnl/ke/thrdobj.c [hunk 1]
```diff
@@ -1137,7 +1139,7 @@ KeSetSystemAffinityThread(IN KAFFINITY Affinity)
         }
 
         /* Calculate the ideal CPU from the affinity set */
-        BitScanReverse(&NodeMask, AffinitySet);
+        BitScanReverseAffinity(&IdealProcessor, AffinitySet);
         CurrentThread->IdealProcessor = (UCHAR)NodeMask;
```
Comment (binarymaster) [inline ntoskrnl/ke/thrdobj.c:1143]: Obtained `IdealProcessor` value being unused?

Comment (tkreuzer) [reply ntoskrnl/ke/thrdobj.c:1143]: Oh, no that should be used instead of `NodeMask`.

File: ntoskrnl/ke/thrdobj.c [hunk 2]
```diff
@@ -1137,8 +1139,8 @@ KeSetSystemAffinityThread(IN KAFFINITY Affinity)
         }
 
         /* Calculate the ideal CPU from the affinity set */
-        BitScanReverse(&NodeMask, AffinitySet);
-        CurrentThread->IdealProcessor = (UCHAR)NodeMask;
+        BitScanReverseAffinity(&IdealProcessor, AffinitySet);
+        CurrentThread->IdealProcessor = (UCHAR)IdealProcessor;
```
Comment (HBelusca) [inline ntoskrnl/ke/thrdobj.c:1143]: Interesting, that the cast is required. I wonder why they used a UCHAR in the TEB (and then of course they upgraded it to a ULONG in windows 7 or 8)

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]

File: ntoskrnl/ke/thrdobj.c (hunk 2 above)
Comment (HBelusca) [reply ntoskrnl/ke/thrdobj.c:1143]: probably ^^

**Review by GeoB99 — APPROVED** [review]

**Pull request merged**